ballerina/modules/soap11/tests/http_soap_service.bal

```diff
@@ -34,29 +35,70 @@ service / on new http:Listener(9090) {
         return response;
     }
 
+    resource function post getMimePayload(http:Request request) returns http:Response|error {
+        http:Response response = new;
+        mime:Entity[] mtomMessage = [];
+        mime:Entity envelope = new;
+        check envelope.setContentType("application/xop+xml");
+        envelope.setContentId("<soap@envelope>");
+        envelope.setBody(check (check request.getBodyParts())[0].getXml());
+        mtomMessage.push(envelope);
+        response.setBodyParts(mtomMessage);
+        response.setPayload(mtomMessage);
+        return response;
+    }
+
     resource function post getSamePayload(http:Request request) returns http:Response|error {
         xml payload = check request.getXmlPayload();
         http:Response response = new;
         response.setPayload(payload);
         return response;
     }
 
+    resource function post getSecuredMimePayload(http:Request request) returns http:Response|error {
+        xml payload = check (check request.getBodyParts())[0].getXml();
+        xml applyOutboundConfig = check soap:applyOutboundConfig(
+            {
+            verificationKey: clientPublicKey,
+            signatureAlgorithm: soap:RSA_SHA256,
+            decryptionAlgorithm: soap:RSA_ECB,
+            decryptionKey: serverPrivateKey
+        }, payload, false);
+        xml securedEnv = check soap:applySecurityPolicies(
+            {
+            signatureAlgorithm: soap:RSA_SHA256,
+            encryptionAlgorithm: soap:RSA_ECB,
+            signatureKey: serverPrivateKey,
+            encryptionKey: clientPublicKey
+        }, applyOutboundConfig, false);
+        http:Response response = new;
+        mime:Entity[] mtomMessage = [];
+        mime:Entity envelope = new;
+        check envelope.setContentType("application/xop+xml");
+        envelope.setContentId("<soap@envelope>");
+        envelope.setBody(securedEnv);
+        mtomMessage.push(envelope);
+        response.setBodyParts(mtomMessage);
+        response.setPayload(mtomMessage);
+        return response;
+    }
+
     resource function post getSecuredPayload(http:Request request) returns http:Response|error {
         xml payload = check request.getXmlPayload();
         xml applyOutboundConfig = check soap:applyOutboundConfig(
             {
-                verificationKey: clientPublicKey,
-                signatureAlgorithm: soap:RSA_SHA256,
-                decryptionAlgorithm: soap:RSA_ECB,
-                decryptionKey: serverPrivateKey
-            }, payload);
+            verificationKey: clientPublicKey,
+            signatureAlgorithm: soap:RSA_SHA256,
+            decryptionAlgorithm: soap:RSA_ECB,
+            decryptionKey: serverPrivateKey
+        }, payload, false);

```

Comment:
Is this formatting correct?

Reply:
Fixed in [14d1d13](https://github.com/ballerina-platform/module-ballerina-soap/pull/112/commits/14d1d137c0db3e9ae9ef1bbf1fbf860c4635fd00)